server/src/main/java/com/thoughtworks/go/server/domain/user/PipelineSelections.java

```diff
@@ -20,159 +20,113 @@
 import com.thoughtworks.go.config.PipelineConfig;
 import com.thoughtworks.go.config.PipelineConfigs;
 import com.thoughtworks.go.domain.PersistentObject;
-import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 import java.io.Serializable;
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Date;
-import java.util.List;
-import java.util.stream.Collectors;
 
 public class PipelineSelections extends PersistentObject implements Serializable {
+    public static final int CURRENT_SCHEMA_VERSION = 1;
 
-    private Date lastUpdate;
-    public static final PipelineSelections ALL = new PipelineSelections() {
-        @Override public boolean includesGroup(PipelineConfigs group) {
+    public static final PipelineSelections ALL = new PipelineSelections(Filters.defaults(), null, null) {
+        @Override
+        public boolean includesGroup(PipelineConfigs group) {
             return true;
         }
 
-        @Override public boolean includesPipeline(PipelineConfig pipeline) {
+        @Override
+        public boolean includesPipeline(CaseInsensitiveString pipelineName) {
             return true;
         }
     };
-    private List<String> pipelines;
+
     private Long userId;
-    private List<CaseInsensitiveString> caseInsensitivePipelineList = new ArrayList<>();
-    private boolean isBlacklist;
+    private Date lastUpdate;
+    private Filters viewFilters = Filters.defaults();
+    private int version;
 
     public PipelineSelections() {
-        this(new ArrayList<>());
+        this(Filters.defaults(), null, null);
+    }
+
+    public PipelineSelections(Filters filters, Date date, Long userId) {
+        update(filters, date, userId);
     }
 
-    public PipelineSelections(List<String> unselectedPipelines) {
-        this(unselectedPipelines, new Date(), null, true);
+    public String getFilters() {
+        return Filters.toJson(this.viewFilters);
     }
 
-    public PipelineSelections(List<String> unselectedPipelines, Date date, Long userId, boolean isBlacklist) {
-        update(unselectedPipelines, date, userId, isBlacklist);
+    public void setFilters(String filters) {
+        this.viewFilters = Filters.fromJson(filters);
+    }
+
+    public Filters viewFilters() {
+        return viewFilters;
+    }
+
+    public int version() {
+        return version;
     }
 
     public Date lastUpdated() {
         return lastUpdate;
     }
 
-    public void update(List<String> selections, Date date, Long userId, boolean isBlacklist) {
+    public void update(Filters filters, Date date, Long userId) {
         this.userId = userId;
-        this.isBlacklist = isBlacklist;
-        updateSelections(selections);
         this.lastUpdate = date;
+        this.viewFilters = null == filters ? Filters.defaults() : filters;
+        this.version = CURRENT_SCHEMA_VERSION;
     }
 
+    @Deprecated // TODO: remove when removing old dashboard
```

Comment:
`includesGroup()`, `includesPipeline()`, and `isBlacklist()` are only used by the old dashboard page. These can be removed once we eliminate the old dashboard. This logic is encapsulated in `DashboardFilter` instances now.

https://github.com/marques-work/gocd/blob/92a6806b3eccb22c0ac7badc9b86dd9c7b613402/server/src/main/java/com/thoughtworks/go/server/domain/user/PipelineSelections.java#L83-L101